addons/html_builder/static/src/core/plugins/save_plugin.js

```diff
@@ -0,0 +1,119 @@
+import { Plugin } from "@html_editor/plugin";
+
+const oeStructureSelector = "#wrapwrap .oe_structure[data-oe-xpath][data-oe-id]";
+const oeFieldSelector = "#wrapwrap [data-oe-field]:not([data-oe-sanitize-prevent-edition])";
+const OE_RECORD_COVER_SELECTOR = "#wrapwrap .o_record_cover_container[data-res-model]";
+const oeCoverSelector = `#wrapwrap .s_cover[data-res-model], ${OE_RECORD_COVER_SELECTOR}`;
+const SAVABLE_SELECTOR = `${oeStructureSelector}, ${oeFieldSelector}, ${oeCoverSelector}`;
+
+export class SavePlugin extends Plugin {
+    static id = "savePlugin";
+    static shared = ["save"];
+
+    resources = {
+        handleNewRecords: this.handleMutations,
+    };
+
+    async save(isTranslation) {
+        const proms = [];
+        for (const fn of this.getResource("before_save_handlers")) {
+            proms.push(fn());
+        }
+        await Promise.all(proms);
+        const saveProms = [...this.editable.querySelectorAll(".o_dirty")].map(async (dirtyEl) => {
+            dirtyEl.classList.remove("o_dirty");
+            const cleanedEl = dirtyEl.cloneNode(true);
+            this.dispatchTo("clean_for_save_handlers", { root: cleanedEl });
+
+            if (isTranslation) {
+                await this.saveTranslationElement(cleanedEl);
+            } else {
+                await this.saveView(cleanedEl);
+            }
+        });
+        await Promise.all(saveProms);
+    }
```

Comment:
We probably want 1 SavePlugin for website builder and 1 for translate website builder